src/services/message.service.ts

```diff
@@ -36,50 +36,51 @@ export class MessageService {
     content: string,
     argOptions: ITextMessageOptions
   ): Promise<void> {
-    const options: ITextMessageOptions = {};
+    const options: ITextMessageOptions = argOptions;
+
+    options.title = argOptions.title?.substring(0, 1000) || '';
     options.header = argOptions.header ? `${argOptions.header}\n` : '';
     options.footer = argOptions.footer ? `\n${argOptions.footer}` : '';
-    options.reply = argOptions.reply || false;
     options.delimiter = argOptions.delimiter === false ? false : argOptions.delimiter || '\n';
 
     const replyLength = options.reply ? `${message.author},  `.length : 0;
     const templateLength = replyLength + options.header.length + options.footer.length;
 
-    let messagesToSend: string[] | null = [`${options.header}${content}${options.footer}`];
+    let messagesToSend: string[] = [`${options.title}${options.header}${content}${options.footer}`];
 
-    if (content.length + templateLength > Constants.MaxMessageLength) {
-      if (!options.delimiter) {
-        return Promise.reject(`Message was over Discords character cap.`);
+    if (messagesToSend.length > Constants.MaxMessageLength) {
+      if (options.dm) {
+        return this._deliverMessageInDm(options.dm, messagesToSend);
       }
+      return this._deliverMessageInChan(message, messagesToSend, !!options.reply);
+    }
 
-      options.delimiter = argOptions.delimiter === '*' ? '' : options.delimiter;
+    if (!options.delimiter) {
```

Comment:
Everything being sent into this function will be hard coded in a plugin, so there should never be an error.

Reply:
Not sure what you are getting at here, since we have proved some plugins can return messages that are too long already.

Reply:
Using this function without a delimiter is literally the same as using the discord `send` function. So it should be required